sycl/plugins/unified_runtime/ur/adapters/hip/common.hpp

```diff
@@ -0,0 +1,176 @@
+//===--------- common.hpp - HIP Adapter -----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+#pragma once
+
+#include <hip/hip_runtime.h>
+#include <ur/ur.hpp>
+
+// Hipify doesn't support cuArrayGetDescriptor, on AMD the hipArray can just be
+// indexed, but on NVidia it is an opaque type and needs to go through
+// cuArrayGetDescriptor so implement a utility function to get the array
+// properties
+inline void getArrayDesc(hipArray *Array, hipArray_Format &Format,
+                         size_t &Channels) {
+#if defined(__HIP_PLATFORM_AMD__)
+  Format = Array->Format;
+  Channels = Array->NumChannels;
+#elif defined(__HIP_PLATFORM_NVIDIA__)
+  CUDA_ARRAY_DESCRIPTOR ArrayDesc;
+  cuArrayGetDescriptor(&ArrayDesc, (CUarray)Array);
+
+  Format = ArrayDesc.Format;
+  Channels = ArrayDesc.NumChannels;
+#else
+#error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
+#endif
+}
+
+// NVidia HIP headers guard hipArray3DCreate behind __CUDACC__, this does not
+// seem to be required and we're not using nvcc to build the UR HIP adapter so
+// add the translation function here
+#if defined(__HIP_PLATFORM_NVIDIA__) && !defined(__CUDACC__)
```

Comment:
Not sure, but I don't think so, I think the comment mentions that this issue happens because we are not using nvcc but doesn't imply that it did exist.